File: w12/d02/Classwork/countingMinutes.js
function countingMinutes(string) {
	var times = string.split("-")
	var start = times[0]
	var finish = times[1]
	var startHour = times[0].split(":")[0]
	var finishHour = times[1].split(":")[0]
	var startMeridian = times[0].match(/["a", "p"]/)[0]
	var finishMeridian = times[1].match(/["a", "p"]/)[0]
	if (startMeridian == "a" && startHour == "12") {
		startHour = "0"
	}
	if (finishMeridian == "a" && finishHour == "12") {
		finishHour = "0"
	}
	var startMinute = start.split(":")[1].match(/\d+/)[0]
	var finishMinute = finish.split(":")[1].match(/\d+/)[0]
	var startInMinutes = (parseInt(startHour) * 60) + parseInt(startMinute)
	var finishInMinutes = (parseInt(finishHour) * 60) + parseInt(finishMinute)

	var diff = 0

	if (startInMinutes > finishInMinutes) {
		diff = (1440 - startInMinutes) + finishInMinutes
	}
	else {
		diff = finishInMinutes - startInMinutes
	}

	console.log(diff)
}

countingMinutes("12:30pm-12:00am")
countingMinutes("12:00am-2:30am")
countingMinutes("1:23am-1:08am")

File: w12/d02/Classwork/counting_minutes.rb
require "pry"

def counting_minutes(string)
	times = string.split("-")
	start = times[0]
	finish = times[1]
	start_hour = start.split(":")[0]
	finish_hour = finish.split(":")[0]
	start_meridian = start.scan(/[a, p]/)[0]
	finish_meridian = finish.scan(/[a, p]/)[0]
	start_hour = "0" if start_hour == "12" && start_meridian == "a"
	finish_hour = "0" if finish_hour == "12" && finish_meridian == "p"
	start_minute = start.split(":")[1].scan(/\d+/)[0]
	finish_minute = finish.split(":")[1].scan(/\d+/)[0]
	start_in_minutes = (start_hour.to_i * 60) + start_minute.to_i
	start_in_minutes += 720 if start_meridian == "p"
	finish_in_minutes = (finish_hour.to_i * 60) + finish_minute.to_i
	finish_in_minutes += 720 if finish_meridian == "p"

	diff = 0
	if start_in_minutes > finish_in_minutes
		diff += (1440 - start_in_minutes) + finish_in_minutes
	else
		diff = finish_in_minutes - start_in_minutes
	end

	puts diff


end

counting_minutes("12:00am-2:30am")

counting_minutes("12:30pm-12:00am")

counting_minutes("1:23am-1:08am")